### PR TITLE
Nextmarker And ListBlobs overload

### PR DESCRIPTION
**Pull request by evjrr**
Return of NextMarker so it is possible to use paging an select more than the first 5000 Blobs

Preformance optimized so collection of 5000 Blobs goes from 45 sec. to  around 9 sec. when using "ABS Container Content" table

Using new overload on ListBlobs will return a dictionary instead of "ABS Container Content" table and the time goes down to 2 sec.

All measurements are based on a local Sandbox Container version 21.5.53619.54870

### CODE REVIEW

**Comment by evjrr** [comment]
@microsoft-github-policy-service agree

File: Modules/System/Azure Blob Services API/src/ABSBlobClient.Codeunit.al [hunk 1]
```diff
@@ -73,6 +73,31 @@ codeunit 9053 "ABS Blob Client"
         exit(ABSClientImpl.ListBlobs(ABSContainerContent, ABSOptionalParameters));
     end;
 
+    /// <summary>
+    /// Lists the blobs in a specific container.
+    /// see: https://go.microsoft.com/fwlink/?linkid=2210588
+    /// </summary>    
+    /// <param name="BlobList">Collection of the result (BlobList of [Text, XmlNode]).</param>
```
Comment (pri-kise) [inline Modules/System/Azure Blob Services API/src/ABSBlobClient.Codeunit.al:80]: Could you please add more details about the BlobList Result. e.g. what is the key of the dictionary.

File: Modules/System/Azure Blob Services API/src/Helper/ABSOptionalParameters.Codeunit.al [hunk 2]
```diff
@@ -334,14 +334,24 @@ codeunit 9047 "ABS Optional Parameters"
     end;
 
     /// <summary>
-    /// Specifies the maximum number of blobs to return
+    /// Specifies the maximum number of blobs to be fetch, from 1 to 5000 if not defined it will fetch 5000
+    /// If more than 5000 blobs then use NextMarker to fetch the next set of blobs
```
Comment (pri-kise) [inline Modules/System/Azure Blob Services API/src/Helper/ABSOptionalParameters.Codeunit.al:338]: Maybe it's worth adding the full details of the documentation to this parameter:

https://learn.microsoft.com/en-us/rest/api/storageservices/list-blobs?tabs=azure-ad#uri-parameters
> In certain cases, the service might return fewer results than specified by maxresults, and also return a continuation token.

> Setting maxresults to a value less than or equal to zero results in error response code 400 (Bad Request).

File: Modules/System/Azure Blob Services API/src/ABSBlobClient.Codeunit.al (hunk 1 above)
Comment (evjrr) [reply Modules/System/Azure Blob Services API/src/ABSBlobClient.Codeunit.al]: Is now updated

File: Modules/System/Azure Blob Services API/src/Helper/ABSOptionalParameters.Codeunit.al (hunk 2 above)
Comment (evjrr) [reply Modules/System/Azure Blob Services API/src/Helper/ABSOptionalParameters.Codeunit.al]: Is now updated

**Review by pri-kise — APPROVED** [review]
looks good to me.
Now maybe someone else to look at it too.

**Comment by JesperSchulz** [comment]
We're on it! Let's get this PR processed!

**Review by JesperSchulz — APPROVED** [review]

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 10731.

**Comment by JesperSchulz** [comment]
Availability update: We will publish a fix for this issue in the next update for release 22.

Build ID to track: 56517.

**Pull request merged**